tests/Loop/DoTest.php

```diff
@@ -121,6 +121,7 @@ function foo(string $b) {}
                             break;
                         }
 
+                        /** @psalm-suppress MixedArgument */
```

Comment:
It makes $a mixed here?

It seemed correctly inferred to `hello` before

Reply:
Yeah, unfortunately that was dodgy too: https://psalm.dev/r/a1a5253dc3. Since assignments in conditionals are a known pain, I thought this was a reasonable compromise (there are many assignment in conditional tests, and this was the only one affected).

Reply:
Seems reasonable